lib/urn.js

```diff
@@ -149,31 +151,26 @@ module.exports = {
 
       if (schemaLength <= 0) {
         throw new Error('Wrong fomatting');
-      }
-
-      if (schemaLength > 255) {
+      } else if (schemaLength > 255) {
         throw new Error('Maximum length of URN schema is 255');
       }
 
-      // Each hex char is 4 bits
-      var idLength = (identity.length - schemaLength - 1) / 2;
-
-      if (idLength != idLength.toFixed()) {
-        throw new Error('Hex chars are of odd count');
-      }
+      var schema = identity.slice(0, schemaLength);
+      var id = identity.slice(schemaLength + 1);
+      var schemaHex = stringToHex(schema);
+      var idHex = stringToHex(id);
+      var idLength = idHex.length / 2;
```

Comment:
not necessary after the conversion to hex is done, it will always be an even length